File: backend/api/EquiApi.Persistence/Model/Saddle.cs
namespace EquiApi.Persistence.Model;

public class Saddle
{
    public int Id { get; set; }
    public required string Name { get; set; }
    
    public int HorseId { get; set; }
    public Horse Horse { get; set; } = null!;
    
    public int CategoryId { get; set; }
    public SaddleCategory Category { get; set; } = null!;
    
    public List<MeasurementGroup> MeasurementGroups { get; set; } = [];
}

public class SaddleCategory
{
    public int Id { get; set; }
    public required string Name { get; set; }
    // TODO: Ask Flora what this is supposed to do; you can always just select the count via sql
    public int Counter { get; set; }
    
    public List<Saddle> Saddles { get; set; } = [];
}

File: reiter_frontend/lib/models/saddle.dart
class Saddle {
  final String id;
  final String name;
  final String category;
  final String horseId;

  Saddle({
    required this.id,
    required this.name,
    required this.category,
    required this.horseId
  });

  factory Saddle.fromJson(Map<String, dynamic> json) {
    return Saddle(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      horseId: json['horseId']
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'horseId' : horseId
    };
  }
}